### PR TITLE
Adds Notifications Filter Bar

### LINKED ISSUE #3241: Notifications: Add Type Filter
Add the same filter at the top of notifications that the web version has:

screen shot 2015-09-24 at 1 25 27 pm

### PR DESCRIPTION
**Pull request by roundhill**
Adds a RadioGroup toolbar at the top of the notifications that allows for filtering of notifications by type.

screen shot 2015-09-28 at 11 51 18 am

Fixes #3241

### CODE REVIEW

**Comment by nbradbury** [comment]
Should there be a custom message for each empty filter view, as there is on the web?

empty-unread-notes

**Comment by nbradbury** [comment]
If I disconnect from wp.com and then switch to the notification tab, I see an empty note list instead of the sign in button. Demo video [here](https://cloudup.com/cmkag-Iu_bJ).

If I force quit the app and start it again, though, the sign in button correctly appears.

**Comment by roundhill** [comment]
> Should there be a custom message for each empty filter view, as there is on the web?

That sounds good to me. @drw158 what are the details with those messages?

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java [hunk 1]
```diff
@@ -243,11 +245,12 @@ public void onClick(View v) {
 
     private void hideEmptyView() {
         if (isAdded() && mEmptyView != null) {
+            mFilterRadioGroup.setVisibility(View.VISIBLE);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.java]: Is this line necessary? The radio group should already be visible.

**Comment by roundhill** [comment]
I just noticed that you can still 'scroll away' the filter bar even when the empty view is showing. Working on a fix for that.

**Comment by drw158** [comment]
I forgot about the empty messages. I think it would be great to add them. I think they are served via the backend (they aren't web-specific) so it should be possible to use them here. There's one for each filter I think.

Related, I think it would be nice to make the link a button. At the time, we couldn't do that on the web. We couldn't have the header, a tagline, and a button. Not sure what changes we would need to do that.

**Comment by roundhill** [comment]
@nbradbury ready for another look, thanks! I've added all of the custom messages for empty filters.

You can easily test empty filters by appending onto this query builder: https://github.com/wordpress-mobile/WordPress-Android/blob/issue/3241-notifications-type-filter/WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.java#L99

with something like this: `.where(Note.Schema.SUBJECT_INDEX, Query.ComparisonType.EQUAL_TO, "foo");`

**Comment by nbradbury** [comment]
The filter bar shows up like this on a Samsung Galaxy S1 emulator running Android 4.1.1:

device-2015-09-30-184539

**Comment by nbradbury** [comment]
I love the new empty messages - really nicely done! This one probably needs to be changed, though, since the reader doesn't provide a way to "search for blogs that share your interests."

device-2015-09-30-185827

**Comment by roundhill** [comment]
> The filter bar shows up like this on a Samsung Galaxy S1 emulator running Android 4.1.1:

Batman approved!

**Comment by roundhill** [comment]
> This one probably needs to be changed, though, since the reader doesn't provide a way to "search for blogs that share your interests."

Ah, good point. I changed it to `Join a conversation: comment on posts from blogs you follow.` cc @drw158

**Comment by nbradbury** [comment]
Looks great! :shipit:

**Pull request merged**